src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2065 @@
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 3][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 27);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 4][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 36);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 4][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 36);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 4][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 36);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 5][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 5][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 5][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 6][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 6][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 6][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 7][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 7][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 7][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 8][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 8][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 8][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 72);
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+__device__ float gaussian(int x, int y, float mulFactor)
+{
+    float expFactor = - ((x * x) + (y * y)) * mulFactor;
+    expFactor = expf(expFactor);
+    float res  = (expFactor * mulFactor) / PI;
+    return res;
+}
+
+__global__ void set_filter_values(float *filterTensor, float *stdDevTensor, int kernelSize, int N, int batchSize)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x);
+    if(id_x >= batchSize)
+        return;
+
+    float *filter = &filterTensor[id_x * N];
+    float stdDev = stdDevTensor[id_x];
+    int cnt = 0;
+    float mulFactor = 1 / (2 * stdDev * stdDev);
+    float kernelSum = 0.0f;
+    for(int i = 0; i < kernelSize; i++)
```

Comment:
For say 9x9 (81 elements), we should ideally be computing expf and gaussian values only for a 5x5 matrix (the top-left quarter 25 elements). Top right quarter is then mirrored to give row0-row4 computed. 
Then bottom half is then copied row by row. Example - 
```
*(d_float9*)row5 = *(d_float9*)row3
*(d_float9*)row6 = *(d_float9*)row2
*(d_float9*)row7 = *(d_float9*)row1
*(d_float9*)row8 = *(d_float9*)row0
```

Reply:
create_gaussian_kernel is a common kernel for all kernel sizes
The above change will vary based on the kernel size

I am able to think of below 2 options which has added in another comment
1. we need to replicate the same code for all kernel sizes inside the if conditions
2. we can have seperate kernels for filter generation based on kernel size

Please let me know your thoughts on this

Reply:
Done